src/samples/coap/lwm2m-client.c

```diff
@@ -337,15 +389,31 @@ del_location_obj(void *instance_data, void *user_data,
     struct sol_lwm2m_client *client, uint16_t instance_id)
 {
     struct location_obj_instance_ctx *instance_ctx = instance_data;
-    bool *has_location_instance = user_data;
+    struct client_data_ctx *data_ctx = user_data;
 
     if (instance_ctx->timeout)
         sol_timeout_del(instance_ctx->timeout);
     free(instance_ctx);
-    *has_location_instance = false;
+    data_ctx->has_location_instance = false;
     return 0;
 }
 
+static void
+bootstrap_cb(void *data,
+    struct sol_lwm2m_client *client,
+    enum sol_lwm2m_bootstrap_event event)
+{
+    if (event == SOL_LWM2M_BOOTSTRAP_EVENT_FINISHED) {
+        printf("...<Call local Bootstrap clean-up operations>...\n"
+            "...<Now that it should have a Server, try to register again>\n");
+        //sol_lwm2m_client_start(client);
```

Comment:
It's not working yet, I need to debug further and find out why. I'm probably messing something related to memory management and losing information... also probably related to your comment about the second `valgrind` error.